unasync.py

```diff
@@ -0,0 +1,287 @@
+                                    renamed_class_call_started = True
+                                    break
+
+                    token = token._replace(src=token_new_src)
+            elif token.name == "STRING":
+                src_token = token.src.replace("'", "")
+                if _STRING_REPLACE.get(src_token) is not None:
+                    new_token = f"'{_STRING_REPLACE[src_token]}'"
+                    token = token._replace(src=new_token)
+                else:
+                    src_token = token.src.replace("\"", "")
+                    if _STRING_REPLACE.get(src_token) is not None:
+                        new_token = f"\"{_STRING_REPLACE[src_token]}\""
+                        token = token._replace(src=new_token)
+
+            new_tokens.append(token)
+            token_counter = token_counter + 1
+
+        return new_tokens
+
+    def _replace_import(self, tokens, token_counter, new_tokens: list):
+        new_tokens.append(tokens[token_counter])
+        new_tokens.append(tokens[token_counter + 1])
+
+        full_lib_name = ''
+        lib_name_counter = token_counter + 2
+        if len(_IMPORTS_REPLACE.keys()) == 0:
+            return lib_name_counter
+
+        while True:
+            if tokens[lib_name_counter].src == " ":
+                break
+            full_lib_name = full_lib_name + tokens[lib_name_counter].src
+            lib_name_counter = lib_name_counter + 1
+
+        for key, value in _IMPORTS_REPLACE.items():
+            if key in full_lib_name:
+                updated_lib_name = full_lib_name.replace(key, value)
+                for lib_name_part in updated_lib_name.split("."):
+                    lib_name_part = self._class_rename(lib_name_part)
+                    new_tokens.append(tokenize_rt.Token("NAME", lib_name_part))
+                    new_tokens.append(tokenize_rt.Token("OP", "."))
+                new_tokens.pop()
+                return lib_name_counter
+
+        lib_name_counter = token_counter + 2
+        return lib_name_counter
+
+    def _class_rename(self, name):
+        if name in _CLASS_RENAME:
+            return _CLASS_RENAME[name]
+        return name
+
+    def _unasync_name(self, name):
+        if name in self.token_replacements:
+            return self.token_replacements[name]
+        return name
+
+
+def unasync_files(fpath_list, rules):
+    for f in fpath_list:
+        found_rule = None
+        found_weight = None
+
+        for rule in rules:
+            weight = rule._match(f)
+            if weight and (found_weight is None or weight > found_weight):
+                found_rule = rule
+                found_weight = weight
+
+        if found_rule:
+            found_rule._unasync_file(f)
+
+
+def find_files(dir_path, file_name_regex):
+    return glob.glob(os.path.join(dir_path, "**", file_name_regex), recursive=True)
+
+
+# Source files ==========================================
+
+_TOKEN_REPLACE["AsyncClient"] = "Client"
```

Comment:
I don't mind so much that this file is a bit complicated, but I would prefer if we could make it simple for developers to make changes to what's renamed/replaced, for example:

```py
# at the top of the file...

class_names = [
  "AblyRest",
  "Push",
  "PushAdmin",
  # etc...
]

# here...
for class_name in class_names:
  _CLASS_RENAME[class_name] = class_name + "Sync"
```

If we do this then if someone adds a new class they will only need to add it to the list instead of having to understand how all this code works :)